enterprise/server/backends/migration_cache/migration_cache.go

```diff
@@ -302,17 +302,33 @@ func (mc *MigrationCache) FindMissing(ctx context.Context, resources []*resource
 		return nil, err
 	}
 
+	if dstErr != nil {
+		log.Warningf("Migration dest FindMissing %v failed: %s", resources, dstErr)
+	}
 	if doubleRead {
-		if dstErr != nil {
-			log.Warningf("Migration dest FindMissing %v failed: %s", resources, dstErr)
-		} else if digest.ElementsMatch(srcMissing, dstMissing) {
+		missingOnlyInDest, missingOnlyInSrc := digest.Diff(srcMissing, dstMissing)
+
+		if len(missingOnlyInSrc) == 0 && len(missingOnlyInDest) == 0 {
 			metrics.MigrationNotFoundErrorCount.With(prometheus.Labels{metrics.CacheRequestType: "findMissingMatch"}).Inc()
 		} else {
 			metrics.MigrationNotFoundErrorCount.With(prometheus.Labels{metrics.CacheRequestType: "findMissing"}).Inc()
-
-			if mc.logNotFoundErrors {
+			if mc.logNotFoundErrors || len(missingOnlyInSrc) != 0 {
 				log.Warningf("Migration FindMissing diff for digests %v: src %v, dest %v", resources, srcMissing, dstMissing)
 			}
+
+			for _, d := range missingOnlyInDest {
+				d := d
+				eg.Go(func() error {
+					mc.sendNonBlockingCopy(ctx, &resource.ResourceName{
```

Comment:
What do you think about adding a parameter to sendNonBlockingCopy to skip the contains check? Then you wouldn't need to fire off goroutines for every digest since you already know they are not in the destination cache and the a plain channel enqueue is very fast.

Reply:
I like that - added

Reply:
Added a couple additional debug logs too